src/quest/extras/dynamodb.py

```diff
@@ -0,0 +1,119 @@
+import os
+from .. import BlobStorage, Blob, WorkflowManager, WorkflowFactory, PersistentHistory, History
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+except ImportError:
+    raise ImportError("The 'dynamodb' extra is required to use this module. Run 'pip install quest-py[dynamodb]'")
+
+class DynamoDBTableCreationException(Exception):
+    pass
+
+class DynamoDB:
+    def __init__(self):
+        self.session = boto3.session.Session(
+            os.environ['AWS_ACCESS_KEY_ID'],
+            os.environ['AWS_SECRET_ACCESS_KEY'],
+            os.environ['AWS_SESSION_TOKEN'],
+            os.environ['AWS_REGION']
+        )
```

Comment:
Do we want to inject the AWS dependency (i.e. the session) or create it here? Would it be better to have a `create_aws_manager()` with some sort of type parameter?